every_election/apps/elections/baker.py

```
@@ -1,2 +1,52 @@
-def push_event_to_queue():
-    print("trigger_rebuild")
+import json
+import logging
+import os
+from typing import Dict, Optional
+
+import boto3
+from django.conf import settings
+
+logger = logging.getLogger(__name__)
+
+
+def send_event(detail: Dict, detail_type: str, source: Optional[str] = None):
+    if not settings.SEND_EVENTS:
+        logger.info(f"Skipping {detail_type} because SEND_EVENTS is disabled.")
+        # don't attempt to push anything in local dev/under test
+        return
+
+    if source is None:
+        source = f"everyelection-{settings.DC_ENVIRONMENT}-{settings.EC2_IP}"
+
+    session = boto3.Session(
+        region_name=os.environ.get("AWS_REGION", "eu-west-2")
+    )
+    events_client = session.client("events")
+
+    try:
+        response = events_client.put_events(
+            Entries=[
+                {
+                    "Source": source,
+                    "DetailType": detail_type,
+                    "Detail": json.dumps(detail),
+                    "EventBusName": settings.DC_EVENTBUS_ARN,
+                }
+            ]
+        )
+        if response["FailedEntryCount"] > 0:
+            logger.error(
+                "Event was unsuccessfully ingested.",
+                extra={"response": response},
+            )
```

What does this mean if the event was "unsuccessfully ingested"? Did we ingest it, or not?
If not, I suggest we say "Event was not ingested." or similar. Lets make it really clear.

---


Side issue: We don't need to deal with this right now, but if we become more dependent on events for propogating changes elsewhere then we probably need to either:

- Have a way store events in EE so we can re-send the event if bad things happen here (or in the the `InternalException` case).
- If we can't push the event to the queue, throw so we rollback the transaction (i.e: we prevent the user from making an edit if we can't also propagate the edit to other systems)

Lets not get bogged down in this issue now though.

---

Yep right now I've taken the approach that it's better to throw an error at sentry, but not block whatever was happening which meant an event was sent. Hopefully the error in [3b2b49f](https://github.com/DemocracyClub/EveryElection/pull/2445/commits/3b2b49f31f99018157dd3a443999a84d7f60d8a5) is clearer. I think 'ingested' came from reading some (tangentially related) docs about cloudwatch logs when I was doing this.